rust/operator-binary/src/config.rs

```diff
@@ -0,0 +1,189 @@
+
+    config.push_str(
+        formatdoc! {r#"
+                # Authentication configs
+                AUTH_TYPE = AUTH_LDAP
+                AUTH_LDAP_SERVER = "{}{}:{}"
+
+                AUTH_LDAP_SEARCH = "{}"
+                AUTH_LDAP_SEARCH_FILTER = "{}"
+                AUTH_LDAP_UID_FIELD = "{}"
+                AUTH_LDAP_GROUP_FIELD = "{}"
+                AUTH_LDAP_FIRSTNAME_FIELD = "{}"
+                AUTH_LDAP_LASTNAME_FIELD = "{}"
+                AUTH_LDAP_EMAIL_FIELD = "{}"
+
+                AUTH_USER_REGISTRATION = "{}"
+                AUTH_USER_REGISTRATION_ROLE = "{}"
+                AUTH_ROLES_SYNC_AT_LOGIN = "{}"
+            "#,
+            match ldap.tls {
+                None => "ldap://",
+                Some(_) => "ldaps://",
+            },
+            ldap.hostname,
+            ldap.port.unwrap_or_else(|| ldap.default_port()),
+            ldap.search_base,
+            ldap.search_filter,
+            ldap.ldap_field_names.uid,
+            ldap.ldap_field_names.group,
+            ldap.ldap_field_names.given_name,
+            ldap.ldap_field_names.surname,
+            ldap.ldap_field_names.email,
+            to_python_bool(
+                authentication_method
+                    .ldap_extras
+                    .as_ref()
+                    .map(|extra| extra.user_registration)
+                    .unwrap_or_else(stackable_superset_crd::default_user_registration)
+            ),
+            authentication_method
+                .ldap_extras
+                .as_ref()
+                .map(|extra| &extra.user_registration_role)
+                .unwrap_or(&stackable_superset_crd::default_user_registration_role()),
+            to_python_bool(
+                authentication_method
+                    .ldap_extras
+                    .as_ref()
+                    .map(|extra| extra.roles_sync_at_login)
+                    .unwrap_or_else(stackable_superset_crd::default_roles_sync_at_login)
+            ),
+        }
+        .as_str(),
+    );
+
+    // Possible TLS options, see https://github.com/dpgaspar/Flask-AppBuilder/blob/master/flask_appbuilder/security/manager.py
+    // app.config.setdefault("AUTH_LDAP_USE_TLS", False)
+    // app.config.setdefault("AUTH_LDAP_ALLOW_SELF_SIGNED", False)
+    // app.config.setdefault("AUTH_LDAP_TLS_DEMAND", False)
+    // app.config.setdefault("AUTH_LDAP_TLS_CACERTDIR", "")
+    // app.config.setdefault("AUTH_LDAP_TLS_CACERTFILE", "")
+    // app.config.setdefault("AUTH_LDAP_TLS_CERTFILE", "")
+    // app.config.setdefault("AUTH_LDAP_TLS_KEYFILE", "")
+    match &ldap.tls {
+        None => config.push_str("AUTH_LDAP_USE_TLS = False\n"),
+        Some(tls) => match &tls.verification {
+            TlsVerification::None {} => {
+                config.push_str(indoc! {r#"
+                    AUTH_LDAP_USE_TLS = False # Strangely we don't want True here because it will use TLS and we need to use SSL.
```

Comment:
I tested it out again and we have to specify `ldaps://` otherwise the connection wont work. So i think its safe to assume TLS is used (and we also set `AUTH_LDAP_TLS_DEMAND = True` when verifying certs). But when we enable `AUTH_LDAP_USE_TLS = False` we get the following error
```
2022-04-07 08:13:39,034:ERROR:flask_appbuilder.security.manager:LDAP Could not activate TLS on established connection with ldaps://openldap.default.svc.cluster.local:1636
```
What would be your preferred way?

Reply:
Ahh, sounds like `AUTH_LDAP_USE_TLS` governs something like STARTTLS (connection starts unencrypted and is then upgraded to TLS) then?

I suspect we'll need to support both STARTTLS and dedicated-TLS-port-mode then :/

Reply:
Done some reading and yeah seems like STARTTLS. For me that would require a change to the Tls struct (e.g. add a field `startTls: bool`). Would it be ok for you to only support dedicated TLS in the first version and add STARTTLS support later?

Reply:
Agreed that this has to go into the common TLS object. I think I'd personally rather have `mode: StartTls | Ldaps`.

Reply:
Aaaaand here we go :D Good thing to split this out into a separate task ;)

Reply:
But i would instinctively put it into the Tls Struct and not making a special field only specific to LDAPs but make it a general Tls setting that not only works for LDAP but also for other protocols (e.g. SMTP or IMAP). But whatever we can discuss that when we implement it

Reply:
Created https://github.com/stackabletech/issues/issues/206 for this. Closing this review comment, please feel free to reopen